src/Files.App/Actions/Content/Archives/Compress/CompressIntoArchiveAction.cs

```diff
@@ -24,7 +24,22 @@ public override async Task ExecuteAsync(object? parameter = null)
 			if (context.ShellPage is null)
 				return;
 
-			var (sources, directory, fileName) = CompressHelper.GetCompressDestination(context.ShellPage);
+			string[] sources = context.SelectedItems.Select(item => item.ItemPath).ToArray();
+			string directory = string.Empty;
+			string fileName = string.Empty;
+
+			if (sources.Length is not 0)
+			{
+				// Get the current directory path
+				directory = context.ShellPage.FilesystemViewModel.WorkingDirectory.Normalize();
+
+				// Get the library save folder if the folder is library item
+				if (App.LibraryManager.TryGetLibrary(directory, out var library) && !library.IsEmpty)
+					directory = library.DefaultSaveFolder;
+
+				// Gets the file name from the directory path
+				fileName = SystemIO.Path.GetFileName(sources.Length is 1 ? sources[0] : directory);
```

Comment:
Can this code be shared across the various actions?

Reply:
Sure, I can put this in the service.

Reply:
Done